src/main/java/com/uid2/shared/store/reader/RotatingClientKeyProvider.java

```diff
@@ -93,4 +95,12 @@ public CloudPath getMetadataPath() {
     public IAuthorizable get(String key) {
         return getClientKey(key);
     }
+
+    public ClientKey getOldestClientKey(int siteId) {
+        List<ClientKey> keys = reader.getSnapshot().values().stream()
+                .filter(k -> k.getSiteId() == siteId) // filter by site id
+                .sorted(Comparator.comparing(ClientKey::getCreated)) // sort by key creation timestamp ascending
+                .collect(Collectors.toList());
```

Comment:
Would `findFirst()` be more appropriate here?

Reply:
Replaced `get(0)` with `findFirst()` as suggested.